Source/Core/Core/NetPlayServer.h

```diff
@@ -103,6 +103,12 @@ class NetPlayServer : public TraversalClientClient
     AllExcept
   };
 
+  enum MappedState
+  {
+    Unmapped,
+    Mapped
+  };
```

Comment:
Didn't catch this before. The only place this enum is used is in the `mapping == Unmapped` check

You can't just define an enum simply because you want to eliminate a constant from your code, it's not authoritative. Worse, this example is wrong, 1 does not mean Mapped. The m_pad_map array actually contains the player ID and zero when there is no player. Comparing an enum to an out-of-range enum might actually be undefined behaviour, though I'm not sure. 

Just keep the 0 constant, there is nothing wrong with it. The rule about "Always declare constants when you can" is a bit of an anti-pattern and can actually hinder readability. 

If you are worried about code readability at the comparison, add a comment saying "zero means no player", but I think anyone reading the code will naturally assume that anyway.

Reply:
It does hinder my the readability when an array is randomly compared to zero. `NULL` would be more appropriate, but I'm not sure how to do that without macros.

Reply:
Although I will take the enum out, I agree that's not appropriate here. I'd love an alternative if you have one though.

Reply:
I disagree, I think 0 is perfectly readable.

But if you really must try and improve readability, this is not the right place for enums. Instead, create a class to wrap the underlying int, allow implicit casing to int so it still works everywhere else. 

Then add some kind of isNull or isEmpty method that does the check with zero internally.

Reply:
Y'know, I don't really like that, but it sounds like the best option for this. I'm going to leave it with the comment for now and if I mess with the pad mapping at some point (or something else in this file), I'll decide if I want to go through with that then. Hopefully, something more illuminating and less wrapping an int in a class will reveal itself.

Reply:
`std::optional` is a generic version of such a class.